src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -12,6 +13,11 @@ namespace Microsoft.PowerFx.Functions
 {
     internal static partial class Library
     {
+        private static bool IsValidDateTimeUO(string s)
+        {
+            return Regex.IsMatch(s, @"^[0-9]{4,4}(-[0-1][0-9](-[0-3][0-9](T[0-9]{2,2}(:[0-9]{2,2}(:[0-9]{2,2}(\.[0-9]{3,3}Z)?)?)?)?)?)?$");
```

Comment:
>.[0-9]{3,3} [](http://example.com/codeflow?start=111&length=11)

If this is optional we should have tests that validate that we support it. Same for all other optional parts.

Reply:
Added tests for optional parts